src/repository/resources.rs

```diff
@@ -20,13 +25,21 @@ pub enum ResourceGetError {
 #[async_trait]
 pub trait ResourceRepository {
     async fn get(&self, project_id: i32) -> Result<Vec<Resource>, ResourceGetError>;
-    async fn insert(&self, project_id: i32, data: &ResourceData)
-        -> Result<(), ResourceInsertError>;
+    async fn get_meta(
+        &self,
+        project_id: i32,
+        resource_id: i32,
+    ) -> Result<Resource, ResourceGetError>;
+    async fn insert(
+        &self,
+        project_id: i32,
+        data: &ResourceMetadata,
+    ) -> Result<Resource, ResourceInsertError>;
     async fn update(
         &self,
         project_id: i32,
         resource_id: i32,
-        data: &ResourceData,
+        content: &[u8],
```

Comment:
Is this array of bytes? (Rust fan curiosity)

Reply:
Yes, more specifically an array slice of bytes